### ISSUE TITLE
One-row matrices with ( ) as fences render as invalid fractions

### ISSUE THREAD

**Issue by xlf1024** [runarberg/ascii2mathml#17]
`(a,b;)` renders to
``` xml
<math display="block">
  <mfenced open="(" close=")">
    <mfrac linethickness="0">
      <mrow><mi>a</mi><mo>,</mo><mi>b</mi></mrow>
    </mfrac>
   </mfenced>
</math>
```
which displays as (invalid markup)
instead of
``` xml
<math display="block">
  <mfenced open="(" close=")">
    <mtable>
      <mtr>
        <mtd><mi>a</mi></mtd>
        <mtd><mi>b</mi></mtd>
      </mtr>
    </mtable>
  </mfenced>
</math>
```

This _only_ happens when using `(`  `)` as _both_ fences, even replacing _one_ of them with `(:` or `:)` avoids the issue. Writing `(;a,b;)`, thus adding an empty row in the beginning, helps too (but slightly alters the rendering).

**Comment by runarberg** [comment]
This seems to happen because I was trying to be smart about the [binomial coefficient](https://en.wikipedia.org/wiki/Binomial_coefficient). I read somewhere that it is preferred to denote the binomial coefficient using `<mfrac>`. But since it is causing this bug, I might just drop it. Perhaps in favor of `binom(a, b)`.

**Comment by xlf1024** [comment]
I think, it would be sufficient to add an extra "empty second row"-check in the binom code, as there are no one-row binomial coefficients and two-row matrices can be already forced by ending the second line with a semicolon.

**Comment by xlf1024** [comment]
I'm not in favor of adding more expressions like `binom(a,b)` as I use ascii2mathml for its brevity and thus speed of typing.

**Comment by runarberg** [comment]
@xlf1024: Well, this bug can easily be fixed, while still translating `(n; k)` as the binomial coefficient. However, I was never sure that was a good idea since two item column vectors are quite common and render slightly different from the binomial coefficient. Think `binom(n, k) + binom(n, k+1)` vs. `(a, b;)(c; d)`. Personally I don’t like it when a software tries to be smarter then me 😃.

This could be annoying for a perfectionist. When they have different sized column vectors in their documents, and for some reason there is this slight difference in the padding for the two item vectors from the rest of them.

![screen shot 2018-02-11 at 10 36 31](https://user-images.githubusercontent.com/3984600/36077038-6d90f120-0f1a-11e8-8e43-d59165ee5703.png)
![screenshot-2018-2-11 mathjax](https://user-images.githubusercontent.com/3984600/36077044-87f60c62-0f1a-11e8-819f-3dd3f7f8ada8.png)

**Above**: The difference between `(n; k)` (left) and `([n], [k])` (right) in ascii2mathml.  
**Below**: The same difference—`$$\binom{n}{k} $$` and `$$\begin{pmatrix}n \\ k \end{pmatrix}$$` respectively—in mathjax.

![screenshot-2018-2-11 ascii2mathml](https://user-images.githubusercontent.com/3984600/36077186-360e0a38-0f1c-11e8-8714-81eae5dd76e6.png)
![screenshot-2018-2-11 ascii2mathml 1](https://user-images.githubusercontent.com/3984600/36077188-3a806624-0f1c-11e8-95cd-5ee00ef03e8c.png)

**Above**: A binomial coefficient and a three item column vector: `(a; b), (a; b; c)`.  
**Below**: A two- and a three item column vectors: `([a], [b]), ([a], [b], [c])`.

**Comment by jonathanPoitz** [comment]
What's the status on this bug @runarberg?, I'm also having the same problem. Imo the behavior should be the same as in incomplete fractions. The input `2/` produces an empty <mrow> while (2;) does not.

**Comment by runarberg** [comment]
I should probably pull in #18. I think that will fix the issue

**Comment by runarberg** [comment]
Fixed in 0.7.0. Thanks for your help @jonathanPoitz and @xlf1024

**Comment by jonathanPoitz** [comment]
Hey @runarberg, thanks for reacting so quickly!
I checked out the code and unfortunately it still doesn't work the way I'm expecting it to.

If I input `2/`, the output is (correctly imo):
`<math><mfrac><mn>23</mn><mrow></mrow></mfrac></math>`

If I input `binom(2,)`, the output however still is:
`<math><mfenced open="(" close=")"><mfrac linethickness="0"><mn>23</mn></mfrac></mfenced></math>` in which the mfrac only has one argument.

Imo, it would make sense if this were behaving identically. Or do you see a reason why it shouldn't?

**Comment by runarberg** [comment]
@jonathanPoitz You are right, `binom` should probably put an empty `<mrow>` if an argument is missing. Let me move that into a new issue

**Comment by runarberg** [comment]
@jonathanPoitz Thanks for the feedback, this is now fixed in 0.7.1
